pkg/middleware/stored_session.go

```diff
@@ -204,8 +193,9 @@ func needsRefresh(refreshPeriod time.Duration, session *sessionsapi.SessionState
 
 // refreshSession attempts to refresh the session with the provider
 // and will save the session if it was updated.
-func (s *storedSessionLoader) refreshSession(rw http.ResponseWriter, req *http.Request, session *sessionsapi.SessionState) error {
-	refreshed, err := s.sessionRefresher(req.Context(), session)
+func (s *storedSessionLoader) refreshSession(rw http.ResponseWriter, req *http.Request, providerMap providers.ProviderMap, session *sessionsapi.SessionState) error {
+
+	refreshed, err := s.providerMap[session.ProviderID].RefreshSession(req.Context(), session)
```

Comment:
We need to get the provider from the map and check it isn't nil, in case there's an invalid ID. Every function that gets it requires this please